lib/remote/jsonrpcconnection.cpp

```diff
@@ -62,13 +62,13 @@ void JsonRpcConnection::HandleIncomingMessages(boost::asio::yield_context yc)
 {
 	m_Stream->next_layer().SetSeen(&m_Seen);
 
-	for (;;) {
+	while (!m_ShuttingDown) {
 		String message;
 
 		try {
 			message = JsonRpc::ReadMessage(m_Stream, yc, m_Endpoint ? -1 : 1024 * 1024);
 		} catch (const std::exception& ex) {
-			Log(m_ShuttingDown ? LogDebug : LogNotice, "JsonRpcConnection")
+			Log(LogNotice, "JsonRpcConnection")
```

Comment:
Acceptable.

Reply:
Wait, couldn't the write loop

1. fail
2. return
3. disconnect
4. set m_ShuttingDown

while JsonRpc::ReadMessage() yields on yc?

So I opt for keeping this as well.